Zend/zend_vm_def.h

```diff
@@ -5999,7 +6010,7 @@ ZEND_VM_HANDLER(181, ZEND_FETCH_CLASS_CONSTANT, VAR|CONST|UNUSED|CLASS_FETCH, CO
 					HANDLE_EXCEPTION();
 				}
 			}
-			if (OP2_TYPE == IS_CONST) {
+			if (!is_constant_deprecated && OP2_TYPE == IS_CONST) {
```

Comment:
```suggestion
			if (OP2_TYPE == IS_CONST && !is_constant_deprecated) {
```
It doesn't really matter here. But with something like `if (foo() && true)` in the specialized version where the lhs of  `&&` can't be proven to be side-effect free it won't get eliminated.

Reply:
Right, I'll amend this